recipes/sdbus-cpp/all/conanfile.py

```diff
@@ -1,19 +1,23 @@
 import os
 
-from conans import ConanFile, CMake, tools
-from conans.errors import ConanInvalidConfiguration
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
+from conan.tools.files import apply_conandata_patches, copy, get, rmdir
+from conan.tools.scm import Version
 
-required_conan_version = ">=1.33.0"
+required_conan_version = ">=1.51.0"
 
 
 class SdbusCppConan(ConanFile):
     name = "sdbus-cpp"
-    license = "LGPL-2.1-or-later"
+    license = "LicenseRef-LGPL-2.1-or-later-WITH-sdbus-cpp-LGPL-exception-1.0"
```

Comment:
Let me explain what is going on with the license here.

The initial version of this recipe was written for sdbus-c++/0.8.3. sdbus-c++/0.8.3 is published under LGPLv2+ ([COPYING](https://github.com/Kistler-Group/sdbus-cpp/blob/v0.8.3/COPYING)). But before publishing sdbus-c++/0.9.0, the author of the library added license exceptions for template code in the header files ([COPYING-LGPL-Exception](https://github.com/Kistler-Group/sdbus-cpp/blob/v0.9.0/COPYING-LGPL-Exception)). The exception is basically a fork of [Qt LGPL exception 1.1](https://spdx.org/licenses/Qt-LGPL-exception-1.1.html) (exact copy of it, but with its own name and versioning).

SPDX specification [says](https://spdx.github.io/spdx-spec/v2.3/SPDX-license-expressions/#d44-exception-with-operator) that a license and an exception may be combined like this: "license-id WITH license-exception-id". In this case the license identifier could be something like "LGPL-2.1-or-later WITH sdbus-c++-LGPL-exception-1.0". But because sdbus-c++-LGPL-exception-1.0 is not a [valid exception identifier](https://spdx.github.io/spdx-spec/v2.3/SPDX-license-list/#a2-exceptions-list) (not one of the well-known exceptions), it is not possible to combine it with a license identifier using 'WITH' operator. So a free-form LicenseRef should be used here, if I understand the SPDX specification correctly.

And now we have the following situation:
* version 0.8.3 is published under LGPL-2.1-or-later
* versions 1.0.0, 1.1.0, 1.2.0 are published under LicenseRef-LGPL-2.1-or-later-WITH-sdbus-cpp-LGPL-exception-1.0

I don't see an option to override license description from the recipe without splitting recipes for different library versions. And I would prefer not to split the recipes, because they will be almost absolutely identical (except the license description). I don't know how to solve it properly, and I would appreciate any advice.

Reply:
Use https://github.com/conan-io/conan-center-index/blob/master/recipes/audiofile/all/conanfile.py#L22  as reference. So you can set according to the version.

Reply:
Oh, cool! I didn't know it is valid to reassign random `ConanFile` fields (e.g. `license`) from `configure` method. I thought it is only applicable to `options`, `settings` and a few other.